core/controllers/acl_decorators.py

```diff
@@ -151,10 +151,16 @@ def test_does_classroom_exist(self, classroom_url_fragment, **kwargs):
             classroom_url_fragment)
 
         if not classroom:
-            _redirect_based_on_return_type(
-                self, '/learn/%s' % constants.DEFAULT_CLASSROOM_URL_FRAGMENT,
-                self.GET_HANDLER_ERROR_RETURN_TYPE)
-            return
+            # This decorator should only be used for JSON handlers, since all
+            # HTML page handlers are expected to be migrated to the Angular
+            # router and access validation for such pages should be done using
+            # the access validation handler endpoint.
+            if self.GET_HANDLER_ERROR_RETURN_TYPE == feconf.HANDLER_TYPE_JSON:
+                raise self.PageNotFoundException
+            else:
+                # As this decorator is not expected to be used with other
+                # handler types, raising a internal error here.
+                raise self.InternalErrorException
```

Comment:
I think normally we don't do it this way. We typically just do "raise Exception(exception_message)" and then it gets caught by the handle_exception clause and processed accordingly. Maybe just do that? Make sure to write a good error message, too.

Reply:
done